docs/design-survey-conditional-logic.md

```diff
@@ -0,0 +1,691 @@
+    // ...
+  }]
+}
+```
+
+- [View Matrix example](https://surveyjs.io/Examples/Library?id=condition-matrixVisibleIf)
+- [View Matrix Dropdown example](https://surveyjs.io/Examples/Library?id=condition-matrixDropdownVisibleIf)
+
+## Conditional Survey Logic (Triggers)
+
+Triggers allow you to implement additional logic that isn't related to [read-only or required state or visibility](#conditional-visibility). Each trigger is associated with an expression and an action. A survey re-evaluates this expression each time values used in it are changed. If the expression returns `true`, the survey preforms the associated action.
+
+The following triggers are available:
+
+- [`complete`](#complete)
+- [`setvalue`](#setvalue)
+- [`copyvalue`](#copyvalue)
+- [`runexpression`](#runexpression)
+  
+---
+
+### `complete`
+
+Completes the survey. The `expression` is evaluated only when a user switches to the next page.
+
+In the following code, a trigger completes the survey if the `"age"` question on this page has a value under 18:
+
+```js
+const surveyJson = {
+  "elements": [{
+    "name": "age", 
+    // ...
+  }],
+  "triggers": [
+    { "type": "complete", "expression": "{age} < 18" }
+  ]
+}
+```
+---
+
+### `setvalue`
+
+Sets a specified value to a given question. The `setValue` property specifies the value; the `setToName` property specifies the question name.
+
+In the following code, triggers are used to set the `"ageType"` value to `"minor"` or `"adult"` based on the `"age"` question value:
+
+```js
+const surveyJson = {
+  "elements": [{
+    "name": "age", 
+    // ...
+  }, {
+    "name": "ageType",
+    // ...
+  }],
+  "triggers": [{
+    "type": "setvalue",
+    "expression": "{age} < 18",
+    "setToName": "ageType",
+    "setValue": "minor"
+  }, {
+    "type": "setvalue",
+    "expression": "{age} >= 18",
+    "setToName": "ageType",
+    "setValue": "adult"
+  }]
+}
+```
+
+---
+
+### `copyvalue`
+
+Copies a value from one question to another. The `fromName` property specifies the source question; the `setToName` property specifies the target question.
+
+In the following code, a trigger copies the `"billingAddress"` question value into the `"shippingAddress"` question if the `"sameAsBilling"` question is `Yes`:
+
+```js
+const surveyJson = {
+  "elements": [{
+    "name": "billingAddress", 
+    // ...
+  }, {
+    "name": "shippingAddress",
+    // ...
+  }, {
+    "name": "sameAsBilling",
+    "choices": [ "Yes", "No" ]
+    // ...
+  }],
+  "triggers": [{
+    "type": "copyvalue",
+    "expression": "{sameAsBilling} = 'Yes'",
+    "fromName": "billingAddress",
+    "setToName": "shippingAddress"
+  }]
+}
+```
+
+---
+
+### `runexpression`
+
+If the `expression` is `true`, runs another expression specified by the `runExpression` property. You can also save the result of `runExpression` as a question value. For this, assign the question's name to the `setToName` property.
```

Comment:
If the `expression` is `true`, the trigger runs another expression specified by the `runExpression` property. You can also save the result of `runExpression` as a question value. For this, assign the question's name to the `setToName` property.